jetpack-mu-wpcom Code block: Improve edit header

sirreal opened the pull request:
## Proposed changes:

- Remove "block control" type interactions from the block header
- Use `PlainText` component for filename
- Move language selection to toolbar (out of header)
- Render block header only when necessary.
- Hide "line number start at" settings if line numbers are not shown.

This also removes a `typography.root` selector. This would split typography into a separate CSS rule with an identical selector. It's removed here to prevent the redundant selector from being created.

### Before

https://github.com/user-attachments/assets/5e610792-a1a1-4d33-acfd-09daccaa9925




### After

https://github.com/user-attachments/assets/80d55753-ee87-4152-97a1-d0bd3bb743ba


### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?


## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
See demo videos.

The block is unreleased and will undergo extensive testing soon.

sirreal committed:
Use PlainText for filename, add block option

sirreal committed:
Move language select out of block into toolbar

sirreal committed:
Fix filename display logic

sirreal committed:
Clean up blockEditor imports

sirreal committed:
changelog

sirreal committed:
Clean up unused styles

sirreal committed:
Style tweaks and improvements

sirreal committed:
Prevent filename display in rendered view if disabled

sirreal committed:
Prevent long filenames from breaking out of block

sirreal committed:
Prevent header expand on language if no language is set

sirreal committed:
Remove unused customSelectLanguageOptions

sirreal committed:
Fix empty language option

sirreal committed:
Clean up Button children

ellatrix reviewed (dismissed):
Haven't tested this myself, but code wise and based on the video this looks like a great improvement.

sirreal committed:
Update panel labels:

- Settings
- Use sentence case everywhere

sirreal committed:
Hide line number start if line numbers are disabled

sirreal committed:
Remove root typography selector

This selector matches the general root so there is
no need to separate it out. It would produce an extra CSS
style rule which is redundant.

sirreal committed:
Clean up void return event handlers
